pkg/provider/server.go

```diff
@@ -0,0 +1,185 @@
+	for k, v := range objects {
+		ov = append(ov, &v1alpha1.ObjectVersion{Id: k, Version: v})
+	}
+	m.objects = ov
+}
+
+// SetFiles sets provider files to return on Mount
+func (m *SpringCloudConfigCSIProviderServer) SetFiles(files []*v1alpha1.File) {
+	var ov []*v1alpha1.File
+	for _, v := range files {
+		ov = append(ov, &v1alpha1.File{
+			Path:     v.Path,
+			Mode:     v.Mode,
+			Contents: v.Contents,
+		})
+	}
+	m.files = ov
+}
+
+// SetProviderErrorCode sets provider error code to return
+func (m *SpringCloudConfigCSIProviderServer) SetProviderErrorCode(errorCode string) {
+	m.errorCode = errorCode
+}
+
+func (m *SpringCloudConfigCSIProviderServer) Start() error {
+	var err error
+	m.listener, err = net.Listen("unix", m.socketPath)
+	if err != nil {
+		return err
+	}
+	go m.grpcServer.Serve(m.listener)
+	return nil
+}
+
+func (m *SpringCloudConfigCSIProviderServer) Stop() {
+	m.grpcServer.GracefulStop()
+}
+
+// Mount implements a provider csi-provider method
+func (m *SpringCloudConfigCSIProviderServer) Mount(ctx context.Context, req *v1alpha1.MountRequest) (*v1alpha1.MountResponse, error) {
+	var attrib Attributes
+	var secret map[string]string
+	var filePermission os.FileMode
+	var err error
+
+	if m.returnErr != nil {
+		return &v1alpha1.MountResponse{}, m.returnErr
+	}
+	if err = json.Unmarshal([]byte(req.GetAttributes()), &attrib); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal attributes, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetSecrets()), &secret); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal secrets, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetPermission()), &filePermission); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal file permission, error: %+v", err)
+	}
+	if len(req.GetTargetPath()) == 0 {
+		return nil, fmt.Errorf("missing target path")
+	}
+
+	err = attrib.verify()
+	if err != nil {
+		return nil, err
+	}
+
+	fileName := fmt.Sprintf("%s-%s.%s", attrib.application, attrib.profile, attrib.fileType)
+	content, err := m.springCloudConfigClient.GetConfig(attrib)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve secrets for %s: %w", fileName, err)
+	}
+	defer content.Close()
+
+	out := &v1alpha1.MountResponse{
+		ObjectVersion: m.objects,
+		Error: &v1alpha1.Error{
+			Code: m.errorCode,
+		},
+		Files: m.files,
+	}
+	file, err := os.OpenFile(path.Join(req.GetTargetPath(), fileName), os.O_RDWR|os.O_CREATE, filePermission)
+	if err != nil {
+		return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+	}
+	defer file.Close()
+
+	_, err = io.Copy(file, content)
+	if err != nil {
+		return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+	}
+	log.Infof("secrets store csi driver mounted %s", fileName)
+	log.Infof("mount point: %s", req.GetTargetPath())
+	//out.Files = append(out.Files, &v1alpha1.File{
```

Comment:
Is this commented code block still needed? I can see that `out.Files` is being set in line 154. So maybe it is not needed.